### PR TITLE
Docs/buffer management

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Documentation on the buffer system.

### CODE REVIEW

File: docs/UsersGuide/best/dynamic-memory.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      client2.allocate -> my_buffer_manager.bufferGetCallee
+      client2.deallocate -> my_buffer_manager.bufferSendIn
+```
+
+The buffer manager should also be hooked up to a rate group used to downlink telemetry and it requires standard
+telemetry, events, and time connections.
+
+**Configuration and Setup**
+
+The number of sub allocations is configured in the `BufferManagerComponentImplCfg.hpp` header.
```
Comment (kevin-f-ortega) [inline docs/UsersGuide/best/dynamic-memory.md:206]: Would be nice to supply the config-param name here.

File: docs/UsersGuide/best/hub-pattern.md [hunk 2]
```diff
@@ -14,3 +14,8 @@ typed ports are connected to typed ports using the Hub as an intermediary to get
 **Figure 9. Hub pattern.** Each hub instance is responsible for connecting to a remote node. Input port calls are
 repeated to corresponding output ports on a remote hub. These hubs have been demonstrated on Sockets,
 ARINC 653 Channels, High-speed hardware buses between nodes, and UARTs between nodes in an embedded system.
+
+## Generic Hub
+
+There is now a standard implementation of the hub pattern. The [GenericHub](../api/c++/html/svc_generic_hub.html) is an
```
Comment (kevin-f-ortega) [inline docs/UsersGuide/best/hub-pattern.md:20]: Is this a valid link?

**Review by kevin-f-ortega — COMMENTED** [review]
Looks good! Just a few minor comments.

File: docs/UsersGuide/best/hub-pattern.md (hunk 2 above)
Comment (LeStarch) [reply docs/UsersGuide/best/hub-pattern.md:20]: Yes, it appears within the generated documentation.  So, the link will become live once this is merged and autodocs run.

File: docs/UsersGuide/best/dynamic-memory.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+```c++
+Fw::Buffer my_buffer = ...;
+U8* const data = my_buffer.getData();
+
+FW_ASSERT(my_buffer.getSize() >= 4); // Prevent overrun on next line
+data[3] = 1;
+```
+
+Full `Fw::Buffer` documentation is available [here](../api/c++/html/class_fw_1_1_buffer.html).
```
[the remark this answers is not in the document]
Comment (LeStarch) [reply docs/UsersGuide/best/dynamic-memory.md:95]: Same as below.  We generate this documentation, so it appears in the published form.

**Comment by LeStarch** [comment]
@kevin-f-ortega, you are awesome.  I commented some, but I need to fix the others.

**Comment by LeStarch** [comment]
@kevin-f-ortega want to check now that I got around to fixing per your comments?

**Review by kevin-f-ortega — APPROVED** [review]
LGTM

**Pull request merged**